bundles/org.openhab.binding.pjlinkdevice/src/main/java/org/openhab/binding/pjlinkdevice/internal/PJLinkDeviceHandler.java

```diff
@@ -0,0 +1,313 @@
+                } else {
+                    logger.warn("Received unknown power command" + command);
+                }
+            }
+            if (channelUID.getId().equals(PJLinkDeviceBindingConstants.CHANNEL_INPUT)
+                    || channelUID.getId().equals(PJLinkDeviceBindingConstants.CHANNEL_INPUT_DYNAMIC)) {
+                if (command == RefreshType.REFRESH) {
+
+                    StringType input = new StringType(device.getInputStatus().getResult().getValue());
+                    updateState(PJLinkDeviceBindingConstants.CHANNEL_INPUT, input);
+                    updateState(PJLinkDeviceBindingConstants.CHANNEL_INPUT_DYNAMIC, input);
+
+                } else {
+                    logger.trace("Received input command" + command);
+                    Input input = new Input(((StringType) command).toString());
+                    device.setInput(input);
+                }
+            }
+
+            boolean isAudioMute = channelUID.getId().equals(PJLinkDeviceBindingConstants.CHANNEL_AUDIO_MUTE);
+            boolean isVideoMute = channelUID.getId().equals(PJLinkDeviceBindingConstants.CHANNEL_VIDEO_MUTE);
+            if (isVideoMute || isAudioMute) {
+                if (command == RefreshType.REFRESH) {
+                    // refresh both video and audio mute, as it's one request
+                    MuteQueryResponseValue muteStatus = device.getMuteStatus();
+                    updateState(PJLinkDeviceBindingConstants.CHANNEL_AUDIO_MUTE,
+                            muteStatus.isAudioMuted() ? OnOffType.ON : OnOffType.OFF);
+                    updateState(PJLinkDeviceBindingConstants.CHANNEL_VIDEO_MUTE,
+                            muteStatus.isVideoMuted() ? OnOffType.ON : OnOffType.OFF);
+                } else {
+                    if (isAudioMute) {
+                        logger.trace("Received audio mute command" + command);
+                        boolean muteOn = command == OnOffType.ON;
+                        device.setMute(MuteInstructionChannel.AUDIO, muteOn);
+                    }
+                    if (isVideoMute) {
+                        logger.trace("Received video mute command" + command);
+                        boolean muteOn = command == OnOffType.ON;
+                        device.setMute(MuteInstructionChannel.VIDEO, muteOn);
+                    }
+                }
+            }
+
+            updateStatus(ThingStatus.ONLINE);
+            logger.trace("Successfully handled command {} on channel {}", command, channelUID.getId());
+        } catch (IOException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+        } catch (ResponseException e) {
+            logger.error(e.getMessage());
+        } catch (AuthenticationException e) {
+            this.handleAuthenticationException(e);
+        }
+
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(PJLinkDeviceConfiguration.class);
```

Comment:
You should first checks the config, like a non empty IP address or a positive TCP port for example. In case a parameter is not valid, you should set the thing status to OFFLINE with ThingStatusDetail.CONFIGURATION_ERROR.

Reply:
Do you know if I can rely on e.g. the min/max attributes defined on the thing parameter? 
<parameter name="tcpPort" type="integer" min="0"  max="65535">

Reply:
No, in case the thing is set in a config file.
But of course the min/max attributes will do the stuff if the user is defining its thing with Paper UI.

Reply:
Thankfully it's possible to reuse the validation procedure from Paper UI, that's what I'm doing now.